### PR TITLE
PA-6488 Added oauth parameters and authentication

### PR DESCRIPTION
**Pull request by SOOS-JAlvarez**
### Changes:
  - Added parameters to be able to authenticate against an OAuth url

**Ticket:** https://soos.atlassian.net/browse/PA-6488

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -84,16 +84,18 @@ def __init__(self):
         # Auth Options
         self.auth_auto: Optional[str] = '0'
         self.auth_loginUrl: Optional[str] = None
-        self.auth_username: Optional[str] = Constants.EMPTY_STRING
```
Comment (SOOS-JAlvarez) [inline main.py:87]: it doesn't make sense to initialize string variables as empty...

File: main.py [hunk 2]
```diff
@@ -326,7 +337,7 @@ def __add_zap_options__(self, args: List[str]) -> NoReturn:
         args.append(" ".join(zap_options))
 
     def __add_custom_option__(self, label, value) -> str:
-        return f"{label}=\"{value}\""
+        return f"{label}='{value}'"
```
Comment (gregjoeval) [inline main.py:340]: nit: remove extra spaces

**Review by gregjoeval — APPROVED** [review]

File: main.py [hunk 3]
```diff
@@ -884,6 +895,23 @@ def parse_args(self) -> None:
...
+            "--oauthTokenUrl",
+            help="Url to authenticate that grants the access_token.",
+            type=str,
+            default=None,
+            required=False
+        )
+
+        parser.add_argument(
+            "--oauthParameters",
+            help="Parameters to be added to the oauth token request. (eg grant_type).",
```
Comment (SOOS-GSteen) [inline main.py:908]: there are usually more than one - maybe show a fake example of what the exact parameter string would be

File: main.py [hunk 4]
```diff
@@ -884,6 +895,23 @@ def parse_args(self) -> None:
             required=False
         )
 
+        parser.add_argument(
+            "--oauthTokenUrl",
+            help="Url to authenticate that grants the access_token.",
```
Comment (SOOS-GSteen) [inline main.py:900]:
```suggestion
            help="The fully qualified authentication URL that grants the access_token.",
```

File: main.py (hunk 3 above)
Comment (SOOS-GSteen) [reply main.py:908]: e.g. "grant_type=authorization_code&scope=openid" (or whatever the form is supposed to be)

Comment (SOOS-JAlvarez) [reply main.py:908]: sure, was thinking of adding to the readme some snippets on different authentications for this purpose on this ticket https://soos.atlassian.net/browse/PA-6608

File: main.py (hunk 4 above)
Comment (SOOS-JAlvarez) [reply main.py]: done

File: main.py (hunk 3 above)
Comment (SOOS-JAlvarez) [reply main.py]: done

**Pull request merged**